rfcs/rfc-bn-messaging-queue.md

```diff
@@ -0,0 +1,313 @@
+			<legalName>Patrick Wei</legalName>
+			<operatingName>ROY&#39;S SOLE PROP</operatingName>
+			<businessTypeCode>01</businessTypeCode>
+			<corporationCertification>
+				<incorporationDate />
+				<incorporationCertificateIdentifier />
+				<jurisdictionCode />
+				<jurisdictionProvinceCode />
+				<jurisdictionCountryCode />
+				<jurisdictionStateCode />
+			</corporationCertification>
+			<businessAddress>
+				<foreignLegacy>
+					<addressDetailLine1>940 BLANSHARD ST</addressDetailLine1>
+					<addressDetailLine2 />
+				</foreignLegacy>
+				<municipality>VICTORIA</municipality>
+				<provinceStateCode>BC</provinceStateCode>
+				<postalCode>V8W3E6</postalCode>
+				<countryCode>CA</countryCode>
+				<returnMailIndicator>0</returnMailIndicator>
+			</businessAddress>
+			<mailingAddress>
+				<foreignLegacy>
+					<addressDetailLine1>940 BLANSHARD ST</addressDetailLine1>
+					<addressDetailLine2 />
+				</foreignLegacy>
+				<municipality>VICTORIA</municipality>
+				<provinceStateCode>BC</provinceStateCode>
+				<postalCode>V8W3E6</postalCode>
+				<countryCode>CA</countryCode>
+				<effectiveDate>2022-08-17</effectiveDate>
+				<returnMailIndicator>0</returnMailIndicator>
+				<careOfLine />
+			</mailingAddress>
+			<clientEvents>
+				<eventCode>03</eventCode>
+				<effectiveDate>2022-08-17</effectiveDate>
+				<expiryDate>9999-12-31</expiryDate>
+			</clientEvents>
+			<relatedBusinessRegistrationNumber />
+			<owners>
+				<ownerIndividualName>
+					<lastName>Wei</lastName>
+					<givenName>Patrick</givenName>
+				</ownerIndividualName>
+				<ownerBN />
+			</owners>
+		</clientBasicInformationSearchResult>
+	</body>
+</SBNClientBasicInformationSearchResponse>
+```
+
+# Approach
+
+1. Introduce a new queue entity-bn. This queue should handle both request based on the queue property. Queue should have
+a business identifier/ business_id and request type (This will be used to identify which api call queue should trigger).
+Once registration filing is processed and business is created then entity-filer emits an event queue 
+(entity.business.bn), then entity-bn inform BC BN Hub about this new registration and once that is processed 
+entity-bn will emit an event queue (entity.business.bn) with request type (get-business-number) to itself, so that it
+can try to get the BN from OneStop (In case of failure it can try multiple times and it has to wait for at least 30
+seconds before making that call). This way we can track which call get failed and no need to send information to CRA
+multiple times. We will have a table to track all the request with request xml and response xml and number of retires.
+
+    ![entity-bn](rfc-bn-messaging/entity-bn.png)
+
+2. Introduce a new job update-business-number. This job should run every 5 minutes. It fetch top 10 pending business
+number request from bn_request table (once the top 10 processed check for the next 10 until there is no pending request)
```

Comment:
Agree with this approach although I am not sure whether the job should run every 5 mins considering the volume of applications we receive in a day . But that can be decided/changed later

Reply:
could you leverage the existing business events subject, with another q listener?
you'd only need to process the CloudEvent that was a registration type.